packages/components/src/popover/types.ts

```diff
@@ -8,7 +8,9 @@ type PositionYAxis = 'top' | 'middle' | 'bottom';
 type PositionXAxis = 'left' | 'center' | 'right';
 type PositionCorner = 'top' | 'right' | 'bottom' | 'left';
 
-type DomRectWithOwnerDocument = DOMRect & {
+type DomRectWithOwnerDocument = ReturnType<
+	Element[ 'getBoundingClientRect' ]
+> & {
```

Comment:
It is. I decided to write `ReturnType<Element['getBoundingClientRect']>` in order to be explicit about the link with the `getBoundingClientRect` function (and to be more coherent with the `VirtualElement` type, which also references the `getBoundingClientRect` function).

If you think that an explicit `DOMRect` makes more sense, I'm happy to make the change

Reply:
I don't think it makes a difference because the former will just resolve to `DOMRect` by TS. I'd prefer `DOMRect` since it's also in the [spec](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect#return_value).

Reply:
Sure! Done in https://github.com/WordPress/gutenberg/pull/43691/commits/6f2a28986b4797ba31c3158a2a8d7e892b96bbb2